examples/macro_reference.rs

```diff
@@ -0,0 +1,197 @@
+use gtk::prelude::{
```

Comment:
It this file used in the book? Do we need it in the book?

Reply:
My original intent was to pull snippets from this file into the text, but I think it ended up being easier to just leave things abstract. Perhaps we should add a link to it, though.